src/app/core/services/fss-popular-search.service.ts

```diff
@@ -40,10 +54,25 @@ export class FssPopularSearchService {
     var date = new Date(currentDate - startDate)
     return date;
   }
+ 
+  getWeekNumber(date: any) {
+      date = new Date(date.valueOf());
+      var dayNumber = (date.getDay() + 3) % 7;
+      date.setDate(date.getDate() - dayNumber - 3);
+      var firstThursday = date.valueOf();
+      date.setMonth(0, 1);
+      if (date.getDay() !== 4)
+        {
+       date.setMonth(0, 1 + ((4 - date.getDay()) + 7) % 7);
+         }
+      var weekNumber = 1 + Math.ceil((firstThursday - date) / 604800000);    
+      return weekNumber;
+  }
 
-  getDateTime(value: any, fssSearchRow: FssSearchRow) {
-    fssSearchRow.value = value.toLocaleDateString('fr-CA');
-    fssSearchRow.time = value.toLocaleTimeString('en-GB',{hour:'2-digit', minute:'2-digit'});
+  getDateTime(value: any) {
+    var date = value.toLocaleDateString('fr-CA');
```

Comment:
Why French Canadian date format?

Reply:
Hi Martin, So while providing a date value to html to display, it should be in 'YYYY-MM-DD' format and French Canadian format uses that particular format. So for, that reason we did 'fr-CA'. Same goes for time value it only accepts in 24-hour format while providing a value to display in HTML that is why we used 'en-GB'.

Reply:
Could we add comments to that effect as somebody else is bound to ask the same question further down the line.

Reply:
thanks @Shrikan14443 , that makes sense now. I do wonder if there is a more explicit way of expressing the intention, but this is fine for now.